aws/hll/aws-s3-transfer-manager/src/download.rs

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+mod discovery;
+mod handle;
+mod header;
+mod object_meta;
+
+use aws_sdk_s3::operation::get_object::builders::{GetObjectFluentBuilder, GetObjectInputBuilder};
+
+use self::object_meta::ObjectMetadata;
+
+pub struct DownloadRequest {
+    pub(crate) input: GetObjectInputBuilder,
+}
+
+// FIXME - should probably be TryFrom since checksums may conflict?
+impl From<GetObjectFluentBuilder> for DownloadRequest {
+    fn from(value: GetObjectFluentBuilder) -> Self {
+        Self {
+            input: value.as_input().clone(),
+        }
+    }
+}
+
+impl From<GetObjectInputBuilder> for DownloadRequest {
+    fn from(value: GetObjectInputBuilder) -> Self {
+        Self { input: value }
+    }
+}
+
+// FIXME - expose common fields?
+pub struct DownloadResponse {
+    /// Object metadata
+    pub object_meta: ObjectMetadata,
+}
```

Comment:
we should expose any of these fields directly if we ever plan to evolve this struct. Additionally, it should be marked with the `#[non_exhaustive]` attribute.